IPython/core/interactiveshell.py

```diff
@@ -440,6 +431,15 @@ def __init__(self, ipython_dir=None, profile_dir=None,
 
         # This is where traits with a config_key argument are updated
         # from the values on config.
+        cfg = kwargs.get('config', {})
+        parent_config = {}
+        parent = kwargs.get('parent', {})
+        if parent:
+            parent_config = getattr(parent, 'config', {})
```

Comment:
Nop, you don't need to pass config, we can "just" pass parent, but you are right that I can just check in self.config which is guarantied to exit. 

```
$ ipython --PromptManager.color_scheme='direct' --InteractiveShell.PromptManager.other='nop' --no-banner
> /Users/bussonniermatthias/dev/ipython/IPython/core/interactiveshell.py(442)__init__()
-> if 'PromptManager' in cfg or 'PromptManager' in parent_config:
(Pdb) self.config
{'PromptManager': {'color_scheme': 'direct'}, 'InteractiveShell': {'PromptManager': {'other': 'nop'}}, 'TerminalIPythonApp': {'display_banner': False}}
(Pdb) parent_config
{'PromptManager': {'color_scheme': 'direct'}, 'InteractiveShell': {'PromptManager': {'other': 'nop'}}, 'TerminalIPythonApp': {'display_banner': False}}
(Pdb) cfg
{}
```